tests/DTO/DTOTest.php

```diff
@@ -74,7 +73,7 @@ public function testIntFrom() {
         $exception = false;
         try {
             DTO::intFrom(["v"=>"0x539"], "v");
-        } catch (\Exception) {
+        } catch (\Exception $e) {
```

Comment:
Let's try and implement exception testing the [PHPUnit](https://phpunit.readthedocs.io/en/9.5/writing-tests-for-phpunit.html#testing-exceptions) way.

Perhaps like this:

```php
$this->expectException(InvalidArgumentException::class);
DTO::intFrom(["v"=>"0x539"], "v");
``` 

See if you spot any other places we are doing this...

Reply:
I recall having some issues with `expectException()` not working the way it's supposed to a few years ago. I can't remember what the issue was, if it was related to PHPUnit, or the way we setup stock2shop/app. So I've avoided using it in general. Maybe that has been fixed now.

Reply:
> I recall having some issues with `expectException()` not working the way it's supposed to a few years ago. I can't remember what the issue was, if it was related to PHPUnit, the PHP version, or the way we setup stock2shop/app. So I've avoided using it in general. Maybe that has been fixed now.

This [article](https://thephp.cc/articles/questioning-phpunit-best-practices) by Sebastian Bergmann seems to suggest that it is the best approach.

Reply:
@flumonion 

Updated the `DTOTest` class, please see method [here](https://github.com/stock2shop/share/blob/ac42b2ec8065220cdffc43c3309480c2e31edad8/tests/DTO/DTOTest.php#L134-L138).

I've replaced all try-catch blocks with a method call:

```php
$this->assertExceptionWithMessage(InvalidArgumentException::class, 'value is non-numeric');
```